### ISSUE TITLE
tabsyn works well in 50 dimensions but cannot be trained to work in 1000 dimensions.

### ISSUE THREAD

**Issue by JunHanStudy** [amazon-science/tabsyn#17]
Thank you for your nice project and codebase!

I have trained Tabsyn on our dataset with binary feature of 1000 dimensions. Most dimensions have very low mean, which means most inputs are zero. The loss for VAE and diffusion are both loss. But the samples look bad. For the same code, I just train it on first 50 dimensions out of 1000 dimensions, the samples looks good. This should indicate that I ran Tabsyn correctly. Do you have an idea why this could happen?

**Comment by JunHanStudy** [comment]
For the same code, I also trained it on first 100 dimensions out of 1000 dimensions, the samples look good. But when I trained it on first 200 dimensions out of 1000 dimensions, the samples start looking bad. Do you have an idea why this could happen? Thank you!

**Comment by hengruizhang98** [comment]
Hi, thanks for your question! I guess this is because your raw data is already extremely sparse, and our VAE processing have made the latent embeddings even sparser (since it expands the dimension of your data from 1000 to 4000). It might be challenging for the diffusion model to learn from such high-dimensional data.

Some questions/suggestions:
1. What's the size of your datasets, i,e., how many rows?
2. Have you tried TabDDPM on your datasets?
3. Considering that all of the columns are binary categorical variables, I would suggest directly training the diffusion model in the input space without VAE.

**Comment by JunHanStudy** [comment]
Thank you very much for your reply!
1. The size of our training data is 1.5M.
2. TabDDPM works well in our dataset of 1000 dimensions. I guess one reason could be that TabDDPM which directly learns the categorical distribution can work well in the sparse setting.
3. We will try this.
As for those columns with sparse features, even all reconstructing zero will have a small reconstruction loss in VAE.

**Comment by hengruizhang98** [comment]
@JunHanStudy 
Have you solved the issue? If you haven't, would you please share (partial) your dataset so that we can do further study about that?

**Comment by JunHanStudy** [comment]
@hengruizhang98 Thank you very much for your follow up! The performance degeneration in high dimension is not solved. We work on healthcare datasets. This link (https://github.com/sczzz3/ehrdiff) has processed MIMIC dataset. Each dimension is a binary variable. Please let me know if you can use this processed dataset easily. If you want this dataset be the format in your paper, I can process it for you. Look forward to seeing some progress of tabsyn in this domain.

**Comment by hengruizhang98** [comment]
Thanks! I will look through it.

**Comment by wxbs** [comment]
Thanks for your nice work! I have trained both Tabsyn and TabDDPM on our dataset with binary feature of more 1000 dimensions. It seems that TabDDPM has a loss of NaN during training. Have you met the problem? 
Your early reply will be highly appreciated! Thanks!

**Comment by hengruizhang98** [comment]
We also faced this issue but failed to find a solution that could perfectly
solve it. Maybe you can try train it for multiple times.

**Comment by wxbs** [comment]
Thanks very much for your early reply! Do you mean that both Tabsyn and TabDDPM have this problem?

**Comment by hengruizhang98** [comment]
We face a similar issue on TabDDPM, while Tabsyn does not have such an
issue.

Hengrui

**Comment by wxbs** [comment]
Thanks very much for your early reply! By the way, d_token in the code of Tabsyn is 4. What is the reason for setting d_token to 4, and do we need to choose different values based on our actual situation?

**Comment by hengruizhang98** [comment]
I tried grid search this hyperparameter and found d = 4 a reasonable value.
A small d makes the VAE encoder less expressive, A large d can makes a
powerful VAE, but the high-dimensional latent space makes the subsequent
diffusion model harder to train, and there is a risk of overfitting in the
VAE model.

Hengrui
